library/general/src/lib/ui/text_helpers.rb

```diff
@@ -20,6 +20,59 @@
 module UI
   # Provides a set of methods to manipulate and transform UI text
   module TextHelpers
+    # Default HTML tags replacements
+    #
+    # @see #plain_text
+    # @return [Hash<String, String>]
+    DEFAULT_HTML_TAGS_REPLACEMENTS = {
+      "<br>"  => "\n",
+      "<br/>" => "\n",
+      "<br >" => "\n",
+      "</li>" => "\n",
+      "<ol>"  => "\n\n",
+      "<ul>"  => "\n\n",
+      "</p>"  => "\n\n"
+    }.freeze
+
+    # Get a new text version after ridding of HTML tags
+    #
+    # By default, a fully plain text will be returned, since some tags (see
+    # {DEFAULT_HTML_TAGS_REPLACEMENTS}) are going to be replaced by line breaks while the rest of
+    # them will be removed.
+    #
+    # However, the _tags_ param allows specifying which tags should be replaced
+    # or removed.  Additionally, a collection of tag => replacement can be
+    # provided via the _replacements_ param.
+    #
+    # Note that all matched tags without a replacement are going to be replaced
+    # with `nil`. In other words, they will be deleted.
+    #
+    # @example Remove all HTML tags
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text) #=> "YaST:\n\na powerful installation and configuration tool."
+    #
+    # @example Removing only the <p> tags
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text, tags: ["p"])
+    #     #=> "YaST:\n\na <b>powerful</b> installation and configuration tool."
+    #
+    # @example Using custom replacements
+    #   text = "<p>YaST:</p><p>a <b>powerful</b> installation and configuration tool.</p>"
+    #   plain_text(text, replacements: { "<p>" => "\n- ", "<b>" => "*", "</b>" => "*" })
+    #     #=> "- YaST:\n- a *powerful* installation and configuration tool."
+    #
+    # @param text [String] text to be processed
+    # @param tags [Array<String>] specific tags to be replaced or deleted
+    # @param replacements [Hash<String, String>] the replacements collection, tag => replacement
+    #
+    # @return [String] the new version after ridding of undesired tags.
+    def plain_text(text, tags: nil, replacements: nil)
+      regex = tags ? Regexp.union(tags.map { |t| /<[^>]*#{t}[^>]*>/i }) : /<.+?>/
+      replacements ||= DEFAULT_HTML_TAGS_REPLACEMENTS
+
+      text.gsub(regex) { |match| replacements[match.to_s.downcase] }.strip
+    end
+
```

Comment:
BTW I think that it would be nice to create refinement for plain_text and wrap_text so you can have really `string_a.plain_text.wrap_text` which looks more nature.

Reply:
Definitely, that's a great idea. Let's try it.

However, we should keep the current #wrap_text version since it has been used in several places.

Reply:
yep, of course backward compatibility is needed.

Reply:
@jreidinger 

Refinements have been added. See https://github.com/yast/yast-yast2/pull/1011/commits/d43265799d1b4231f6c1a2a60f56745e31046240

However, below problems were found

* [YARD](https://yardoc.org/features.html) does not generate the documentation. See this old and closed [issue](https://github.com/lsegal/yard/issues/780). I did tests and the workaround kind of work, but the file looks weird.
* I don't know what to do with tests. Those refinements are already tested via the helpers.
* I'm not sure if the _library/general/src/lib/yast2/refinements/_ was a good choice.

Could you give me a hand, please?

Thanks a lot!